### PR TITLE
Homepage

### PR DESCRIPTION
**Pull request by vidushit09**
Added css and html files for home page
Screenshot 2023-02-03 at 4 51 55 PM

Screenshot 2023-02-03 at 4 19 44 PM

https://user-images.githubusercontent.com/123439039/216988455-96beceae-d107-4ffc-bac8-631b84735be0.mov

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -0,0 +1,405 @@
...
+        </div>
+        <div class="rightNav">
+            <div class="login">Login</div>
+            <button class="myCart">
+                <i class='fa fa-shopping-cart'></i>My Cart
+            </button>
+        </div>       
+    </div>
+    <ul class="list">
+        <li style="margin-left: 0vw;">Vegetables & Fruits</li>
```
Comment (dk1710) [inline index.html]: 1. I see that you have used `vw` for responsive page, but for now replace all of that with `rem`. Check why `vw` is not always the good idea
2. Move all these inline styles to css file for consistency.

File: index.html [hunk 2]
```diff
@@ -0,0 +1,405 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+    <title>Document</title>
+</head>
+<body>
+    <div class="navbar">
+        <div class="leftNav">
+            <div class="logo">
```
Comment (dk1710) [inline index.html]: Check if this `div` is required or not

File: index.html [hunk 3]
```diff
@@ -0,0 +1,405 @@
...
+        <div class="rightNav">
+            <div class="login">Login</div>
+            <button class="myCart">
+                <i class='fa fa-shopping-cart'></i>My Cart
+            </button>
+        </div>       
+    </div>
+    <ul class="list">
+        <li style="margin-left: 0vw;">Vegetables & Fruits</li>
+        <li> Dairy & Breakfast</li>
```
Comment (dk1710) [inline index.html]: Why there is a space before `Dairy` ?

File: index.html [hunk 4]
```diff
@@ -0,0 +1,405 @@
...
+            </div>
+        </div>
+        <div class="rightNav">
+            <div class="login">Login</div>
+            <button class="myCart">
+                <i class='fa fa-shopping-cart'></i>My Cart
+            </button>
+        </div>       
+    </div>
+    <ul class="list">
```
Comment (dk1710) [inline index.html]: rename `list`

File: index.html [hunk 5]
```diff
@@ -0,0 +1,403 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+    <title>Document</title>
```
Comment (dk1710) [inline index.html]: Let's give BlinkIt

File: index.html [hunk 6]
```diff
@@ -0,0 +1,403 @@
...
+                <div class="deliveryTime">
+                    Delivery in 10 minutes
+                </div>
+                <div class="location">
+                    Delhi, India <i class='fa fa-caret-down'></i>
+                </div>
+            </div>
+        </div>
+        <div class="rightNav">
+            <div class="login">Login</div>
```
Comment (dk1710) [inline index.html]: It will be a button

File: index.html [hunk 7]
```diff
@@ -0,0 +1,403 @@
...
+        <li>Cold Drinks & Juices</li>
+        <li>Instant & Frozen Food</li>
+        <li>Tea, Coffee & Health Drinks</li>
+        <li>Bakery & Biscuits</li>
+        <li>More <i class='fa fa-angle-down'></i></li>
+    </ul>
+    <div class="productsContainer">
+        <div class="productCategory">
+            <li class="active" style="border-left: 0.4vw solid rgb(36 150 64);">
+                <div class="image">
```
Comment (dk1710) [inline index.html]: 1. rename `image`
2. Images are not coming in the specified box

File: index.html [hunk 8]
```diff
@@ -0,0 +1,403 @@
...
+                            </div>
+                        </div>
+                        <button>ADD</button>
+                    </div>
+                </div>
+            </div> 
+        </div>
+
+    </div>
+    <div class="cards">
```
Comment (dk1710) [inline index.html]: rename `cards`

File: style.css [hunk 9]
```diff
@@ -0,0 +1,375 @@
+body{
+    margin: 0rem;
+    padding: 0rem;
+}
+:root{
+    font-size: 17px;
```
Comment (dk1710) [inline style.css]: 10px

File: index.html [hunk 10]
```diff
@@ -0,0 +1,403 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+    <title>Document</title>
+</head>
+<body>
+    <div class="navbar">
+        <div class="leftNav">
+            <img src="img/logo.png" alt="" class="logo">
```
Comment (dk1710) [inline index.html]: Let's just add text for logo

File: index.html [hunk 11]
```diff
@@ -0,0 +1,442 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+    <title>Blinkit</title>
+</head>
+<body>
+    <div class="navbar">
+        <div class="leftNav">
+            <span class="blink">blink</span>
```
Comment (dk1710) [inline index.html]: wrap this blinkit with div because this should be rendered as combined value. If you change the flex direction of  `leftNav` to columns then it will break

File: style.css [hunk 12]
```diff
@@ -0,0 +1,387 @@
...
+    height: 8.05rem;
+    border-right: 0.12rem solid #eeee;
+}
+.blink{
+    margin-left: 3.5rem;
+    color: #f8cb46;
+    font-size: 4.5rem;
+    font-family: inherit;
+    letter-spacing: -0.3rem;
+    font-weight: bold;
```
Comment (dk1710) [inline style.css:27]: you can move ` font-size: 4.5rem;
    font-family: inherit;
    letter-spacing: -0.3rem;
    font-weight: bold;` to `logo` class

File: style.css [hunk 13]
```diff
@@ -0,0 +1,387 @@
...
+    justify-content: space-between;
+    border-bottom: 0.12rem solid #eee;
+    font-family: Arial, Helvetica, sans-serif;
+}
+.logo{
+    height: 8.05rem;
+    border-right: 0.12rem solid #eeee;
+}
+.blink{
+    margin-left: 3.5rem;
```
Comment (dk1710) [inline style.css]: this margin should be in `logo` styles

File: style.css [hunk 14]
```diff
@@ -0,0 +1,387 @@
...
+    display: flex;
+    justify-content: space-between;
+    border-bottom: 0.12rem solid #eee;
+    font-family: Arial, Helvetica, sans-serif;
+}
+.logo{
+    height: 8.05rem;
+    border-right: 0.12rem solid #eeee;
+}
+.blink{
```
Comment (dk1710) [inline style.css]: blink <> greenColor

File: style.css [hunk 15]
```diff
@@ -0,0 +1,387 @@
...
+    margin-left: 3.5rem;
+    color: #f8cb46;
+    font-size: 4.5rem;
+    font-family: inherit;
+    letter-spacing: -0.3rem;
+    font-weight: bold;
+}
+.it{
```
Comment (dk1710) [inline style.css]: it <> yellowColor

File: style.css [hunk 16]
```diff
@@ -0,0 +1,387 @@
...
+    font-family: inherit;
+    letter-spacing: -0.3rem;
+    font-weight: bold;
+}
+.leftNav{
+    display: flex;
+    align-items: center;
+}
+.rightNav{
+    display: flex;
```
Comment (dk1710) [inline style.css:41]: `display: flex;
    align-items: center;`

Above style is duplicated, please create separate class for this style and reuse that class

File: style.css [hunk 17]
```diff
@@ -0,0 +1,387 @@
...
+    align-items: center;
+}
+.deliveryDetails{
+    margin-left: 4.477rem;
```
Comment (dk1710) [inline style.css]: We don't need to give sizes in this complex number. You can round it off to `4.5rem`. It won't make much difference. Please check at other places also

File: style.css [hunk 18]
```diff
@@ -0,0 +1,387 @@
...
+.myCart:hover{
+    cursor: pointer;
+}
+.myCart i{
+    font-size: 1.677rem;
+    margin-right: 0.8955rem;
+}
+.categoryList{
+    margin: 0rem;
+    padding: 0rem;
```
Comment (dk1710) [inline style.css:93]: `margin: 0rem;
    padding: 0rem;`

is this required?

File: style.css [hunk 19]
```diff
@@ -0,0 +1,387 @@
...
+    font-family: Arial, Helvetica, sans-serif;
+    color: #494545;
+    height: 5.97rem;
+    border-bottom: 0.1492rem solid #d6d6d6;
+    list-style: none;
+    letter-spacing: 0.07462rem;
+    color: #6a6868;
+}
+.categoryList:first-child{
+    margin-left: 0rem;
```
Comment (dk1710) [inline style.css]: This hack shouldn't be needed, try some other approach

File: index.html [hunk 20]
```diff
@@ -0,0 +1,419 @@
...
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <link rel="stylesheet" href="style.css">
+    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/4.7.0/css/font-awesome.min.css">
+    <title>Blinkit</title>
+</head>
+<body>
+    <div class="navbar display-flex">
+        <div class="navbar__left display-flex">
+            <div class="navbar__blinkit display-flex">
```
Comment (dk1710) [inline index.html]: `navbar__blinkit` <> `navbar__logo`

File: style.css [hunk 21]
```diff
@@ -0,0 +1,359 @@
...
+    display: flex;
+    align-items: center;
+}
+.navbar__blinkit{
+    border-right: 0.12rem solid #eeee;
+    font-size: 4.5rem;
+    font-family: inherit;
+    letter-spacing: -0.3rem;
+    font-weight: bold;
+    height: 8rem;
```
Comment (dk1710) [inline style.css]: Why is this required?

File: style.css [hunk 22]
```diff
@@ -0,0 +1,359 @@
...
+    padding-right: 3rem;
+    padding-left: 3rem;
+}
+.navbar__blink-yellow{
+    color: #f8cb46;  
+}
+.navbar__it-green{
+    color: #54b226;
+}
+.navbar__left{
```
Comment (dk1710) [inline style.css]: duplicate style, you can use `display-flex` class

File: style.css [hunk 23]
```diff
@@ -0,0 +1,359 @@
...
+    color: #494545;
+    font-weight: lighter;
+    font-size: 1.67rem;
+    padding: 1.5rem;
+    border: none;
+    background: white;
+}
+.navbar__login:hover{
+    border-radius: 0.5rem;
+    padding: 1.5rem;
```
Comment (dk1710) [inline style.css]: Is this required?

File: style.css [hunk 24]
```diff
@@ -0,0 +1,352 @@
...
+    padding: 0rem;
+    justify-content: center;
+    font-size: 1.51rem;
+    font-family: Arial, Helvetica, sans-serif;
+    color: #494545;
+    height: 6rem;
+    border-bottom: 0.15rem solid #d6d6d6;
+    list-style: none;
+    letter-spacing: 0.07rem;
+    color: #6a6868;
```
Comment (dk1710) [inline style.css:92]: color is defined two times

File: index.html [hunk 25]
```diff
@@ -0,0 +1,419 @@
...
+        </div>
+        <div class="navbar__right">
+            <button class="navbar__login">Login</button>
+            <button class="navbar__mycart">
+                <i class='fa fa-shopping-cart'></i>My Cart
+            </button>
+        </div>       
+    </div>
+    <ul class="category-list display-flex">
+        <li class="category-list__li display-flex">Vegetables & Fruits</li>
```
Comment (dk1710) [inline index.html]: `category-list__li` <> `category-list__list-item`

File: index.html [hunk 26]
```diff
@@ -0,0 +1,419 @@
...
+        <li class="category-list__li display-flex">Munchies</li>
+        <li class="category-list__li display-flex">Cold Drinks & Juices</li>
+        <li class="category-list__li display-flex">Instant & Frozen Food</li>
+        <li class="category-list__li display-flex">Tea, Coffee & Health Drinks</li>
+        <li class="category-list__li display-flex">Bakery & Biscuits</li>
+        <li class="category-list__li display-flex">More <i class='fa fa-angle-down'></i></li>
+    </ul>
+    <div class="products-container">
+        <ul class="products-container__category">
+            <li class="products-container__active display-flex">
```
Comment (dk1710) [inline index.html]: `products-container__active` <> `products-container__category__active`

File: index.html [hunk 27]
```diff
@@ -0,0 +1,419 @@
...
+                <div class="products-container__product-image">
```
Comment (dk1710) [inline index.html]: `products-container__product-image` <> `products-container__category__product-image`

File: index.html [hunk 28]
```diff
@@ -0,0 +1,419 @@
...
+            </li>
+        </ul>
+        <div class="products-container__items-container">
+            <div class="products-container__items-header display-flex">
+                <div class="products-container__buy-fresh-online">Buy Fresh Vegetables Online</div>
+                <div class="products-container__items-header-right display-flex">
+                    <div class="products-container__sort">
+                        Sort By
+                    </div>
+                    <div class="products-container__dropdown">
```
Comment (dk1710) [inline index.html]: Is this div required?

File: index.html [hunk 29]
```diff
@@ -0,0 +1,419 @@
...
+                    </div>
+                    
+                    <div class="products-container__item-name">
+                        New Potato (Aloo)
+                    </div>
+                    <div class="products-container__item-weight">
+                        1 kg
+                    </div>
+                    <div class="products-container__item-footer">
+                        <div class="products-container__price">
```
Comment (dk1710) [inline index.html]: `products-container__price` <> `products-container__price-details`

File: style.css [hunk 30]
```diff
@@ -0,0 +1,352 @@
...
+}
+.category-list__li:hover{
+    cursor: pointer;
+    background-color: #8080801c
+}
+
+.products-container{
+    width: 111.94rem;
+    margin: auto;
+    display: grid;
```
Comment (dk1710) [inline style.css]: Why `grid` for `products-container` ?

File: style.css [hunk 31]
```diff
@@ -0,0 +1,352 @@
...
+    margin: 0rem;
+    padding: 0rem;
+}
+.products-container__product-image{
+    height: 4.48rem;
+    width: 4.48rem;
+    border-radius: 0.75rem;
+    border: none;
+    background-color: #f8f8f8;
+    display: flex;
```
Comment (dk1710) [inline style.css:120]: Why flex is required?

Comment (vidushit09) [reply style.css:120]: In the image div, the image is placed at the bottom center of the outer div. To achieve that I used display:flex. I could have used display_flex class as well which I have used in the rest of the elements, but here we needed align-items:end instead of align-items: center (which is defined in the display_flex class). So to avoid overwriting I used display:flex in this class itself.

File: style.css [hunk 32]
```diff
@@ -0,0 +1,352 @@
...
+    border: 0.07rem solid #d6d6d6;
+}
+.products-container__item-image{
+    position: relative;
+    height: 13rem;
+}
+
+.products-container__item-image img{
+    width: 70%;
+    margin-left: 2.58rem;
```
Comment (dk1710) [inline style.css]: can we use `margin: auto` ?

File: style.css [hunk 33]
```diff
@@ -0,0 +1,352 @@
...
+}
+.products-container__discount{
+    width: 5rem;
+    background: rgb(37, 111, 239);
+    color: white;
+    padding: 0.22rem 0rem 0.22rem 0.22rem;
```
Comment (dk1710) [inline style.css]: Why right padding is zero?
How are you positioning this element? I don't see any left or top property

File: index.html [hunk 34]
```diff
@@ -0,0 +1,419 @@
...
+                </div> 
+
+                
+            </div> 
+        </div>
+
+    </div>
+    <div class="advertisement-cards display-flex">
+        <div class="advertisement-cards__card display-flex">
+            <div class="advertisement-cards__img">
```
Comment (dk1710) [inline index.html]: Is this div required?

File: style.css [hunk 35]
```diff
@@ -0,0 +1,352 @@
...
+}
+.advertisement-cards__title{
+    font-weight: bold;
+    font-size: 1.5rem;
+    margin-bottom: 2.24rem;
+}
+.advertisement-cards__content{
+    font-size: 1.51rem;
+    color: gray;
+    align-items: center;
```
Comment (dk1710) [inline style.css]: Why is this required?

File: style.css [hunk 36]
```diff
@@ -0,0 +1,352 @@
...
+    width: 122.4rem;
+    margin: auto;
+    margin-top: 6rem;
+    padding-bottom: 4.48rem;
+    border-bottom: 0.12rem solid #eeeeee;
+    margin-bottom: 3rem;
+}
+.footer{
+    background-color: #fcfcfc;
+    padding: 0rem 13.43rem;
```
Comment (dk1710) [inline style.css:294]: Why are we not giving width and margin here?

File: style.css [hunk 37]
```diff
@@ -0,0 +1,352 @@
...
+    width: 40.3rem;
+    color: #666666;
+}
+.footer__download-app{
+    font-size: 1.68rem;
+    font-weight: bold;
+    color: #666666;
+    letter-spacing: 0.075rem;
+}
+.footer__app-store, .footer__play-store{
```
Comment (dk1710) [inline style.css]: check if we can give some common name instead of two different class name

File: style.css (hunk 36 above)
Comment (vidushit09) [reply style.css:294]: Footer background has a slightly different colour than the page background. If we give margin, we won't get the background colour throughout the footer.

File: style.css (hunk 33 above)
Comment (vidushit09) [reply style.css]: I did not use the left or top property because the `products-container__discount` div is the first in the `products-container__item-image`. So, naturally it was taking the `left:0rem` and `top:0rem` values. Setting `position:absolute` allows the discount to overlay the image.

File: style.css [hunk 38]
```diff
@@ -0,0 +1,339 @@
...
+    display: flex;
+    align-items: center;
+}
+.navbar__logo{
+    border-right: 0.12rem solid #eeee;
+    font-size: 4.5rem;
+    letter-spacing: -0.3rem;
+    font-weight: bold;
+    padding-right: 3rem;
+    padding-left: 3rem;
```
Comment (dk1710) [inline style.css]: Since both value of padding are same, we can use `padding: 0 3rem`

File: index.html [hunk 39]
```diff
@@ -0,0 +1,411 @@
...
+        <li class="category-list__list-item display-flex">Cold Drinks & Juices</li>
+        <li class="category-list__list-item display-flex">Instant & Frozen Food</li>
+        <li class="category-list__list-item display-flex">Tea, Coffee & Health Drinks</li>
+        <li class="category-list__list-item display-flex">Bakery & Biscuits</li>
+        <li class="category-list__list-item display-flex">More <i class='fa fa-angle-down'></i></li>
+    </ul>
+    <div class="products-container">
+        <ul class="products-container__category">
+            <li class="products-container__category--active display-flex">
+                <div class="products-container__category__product-image">
```
Comment (dk1710) [inline index.html]: `products-container__category__product-image` <> `products-container__category__category-image`

File: style.css [hunk 40]
```diff
@@ -0,0 +1,339 @@
...
+    border-bottom: 0.12rem solid #eee;
+    border-left: 0.6rem solid white;
+}
+
+.products-container__category li:hover:not(.products-container__category--active){
+    background-color: rgb(235, 255, 239);
+    border-left: 0.6rem solid rgb(235, 255, 239);
+    cursor: pointer;
+}
+.products-container__category .products-container__category--active{
```
Comment (dk1710) [inline style.css]: Do we need to add `products-container__category` class here?
